### PR TITLE
Modified insertions and joins for empty matrix

### PR DESCRIPTION
**Pull request by aravindkanna**
This solves the issue [10770](https://github.com/sympy/sympy/issues/10770). I think this is the best and simplest way to solve this. This just returns the argument matrix which we want to insert or join irrespective of the position which is the desired thing.

### CODE REVIEW

**Comment by smichr** [comment]
This should reference #10781. A better design issue there is that the same type of matrix is returned as was represented by the empty matrix at the start.

**Comment by aravindkanna** [comment]
@smichr 

Sir, I didn't get you. Can you please elaborate?

File: sympy/matrices/tests/test_matrices.py [hunk 1]
```diff
@@ -2082,6 +2084,9 @@ def test_row_insert():
         l = [1, 0, 0]
         l.insert(i, 4)
         assert flatten(eye(3).row_insert(i, r4).col(0).tolist()) == l
+    a = Matrix([1, 2])
+    assert Matrix([]).row_insert(10, a) == a
```
Comment (smichr) [inline sympy/matrices/tests/test_matrices.py]: I think for safety reasons, this should raise an error. If someone thinks they are working with a matrix that has size and they aren't, the error should not pass silently. Though a case could be made that list slices that contain out of range bounds are silently truncated as in `L = [0, 1, 2]; L[:10] = [42]` -> `L == [42]`. Do you see any merit in either approach?

**Comment by aravindkanna** [comment]
Sir, why is it saying checks have failed. I have checked on my system before committing. Then if there are any errors, I think it would say *DO NOT COMMIT*. But it didn't say anything. So I have committed. What can I do now Sir?

**Comment by jksuom** [comment]
The checks have failed because the support for Python versions 2.6 and 3.2 was dropped a few days ago.
You should probably pull and merge a new version of master into your branch.

**Comment by aravindkanna** [comment]
@smichr , 
Sir, here my main aim is to rebuild a matrix back from an empty matrix. For that we didn't have any routines till now. But as you pointed a developer must be aware that he is working with an empty matrix, ,this may not deal that issue. I think this is an intended one. There is no need of extra functionality. For example

`a = Matrix([1, 2])`
`a.col_del(0)`
`c = Matrix([3, 4])`
`if not a:`
`   b = c`
`else:`
`   b = a.col_insert(1, c)`

This example looks like it is sufficient and no need of extra functionality.

**Comment by aravindkanna** [comment]
Thank you, @jksuom . Have done the same :smile:

File: sympy/matrices/matrices.py [hunk 2]
```diff
@@ -3983,6 +3983,8 @@ def row_join(self, rhs):
         row
         col_join
         """
+        if not self:
+            return rhs
```
Comment (smichr) [inline sympy/matrices/matrices.py]: here and below, this should be `return self.func(rhs)` (or similar) so the same type of matrix represented by self is returned. But I think the other duplicate PR will be doing this, so let's see.

**Comment by aravindkanna** [comment]
@smichr 
I have made changes to the necessary functions in `matrices.py` similarly. 
`if not self:`
   ` typ = type(self)`
   ` return typ(otherargument) `

Then what I have found is that I also have to change `sparse.py` for sparse matrices. So I changed that too. Then I have checked for every combination. It is returning the type of matrix on which we apply the function.

Thanks for the suggestion :+1: :smile:

**Comment by aravindkanna** [comment]
`No output has been received in the last 10 minutes, this potentially indicates a stalled build or something wrong with the build itself.`

Sir it says this. What type of error is this?

**Comment by smichr** [comment]
> What type of error is this?

I think there is a slow test function that sometimes takes too long to give output so the build stops. I'll restart it.

**Comment by aravindkanna** [comment]
@smichr 
Sir, thanks a lot. All checks have passed. :smile:

**Comment by smichr** [comment]
The other PR makes most of these changes but it doesn't include the SparseMatrix mods. I left a note for you at #10781; would you be willing to make those modifications?

**Comment by aravindkanna** [comment]
@smichr , 
Yeah, sure. Now I will pull the master again. Make necessary changes for sparsematrix or other if they were not implemented on the other PR. I will commit again making the necessary changes. So am I allowed to do it?

**Comment by aravindkanna** [comment]
Also it is saying merge conflicts. So I will drop this branch. I will pull again from master and will push from the same branch.

**Comment by aravindkanna** [comment]
Sir, I don't know what is happening. I am creating a new PR for only the modification of `sparse.py`.

**Pull request closed without merging**